src/virtualship/utils.py

```diff
@@ -42,37 +43,33 @@ def _generic_load_yaml(data: str, model: BaseModel) -> BaseModel:
     return model.model_validate(yaml.safe_load(data))
 
 
-def mfp_to_yaml(excel_file_path: str, yaml_output_path: str):  # noqa: D417
-    """
-    Generates a YAML file with spatial and temporal information based on instrument data from MFP excel file.
+def load_coordinates(file_path):
+    """Loads coordinates from a file based on its extension."""
+    if not os.path.isfile(file_path):
+        raise FileNotFoundError(f"File not found: {file_path}")
 
-    Parameters
-    ----------
-    - excel_file_path (str): Path to the Excel file containing coordinate and instrument data.
+    ext = os.path.splitext(file_path)[-1].lower()
 
-    The function:
-    1. Reads instrument and location data from the Excel file.
-    2. Determines the maximum depth and buffer based on the instruments present.
-    3. Ensures longitude and latitude values remain valid after applying buffer adjustments.
-    4. returns the yaml information.
+    try:
+        if ext in [".xls", ".xlsx"]:
+            return pd.read_excel(file_path)
 
-    """
-    # Importing Schedule and related models from expedition module
-    from virtualship.expedition.instrument_type import InstrumentType
-    from virtualship.expedition.schedule import Schedule
-    from virtualship.expedition.space_time_region import (
-        SpaceTimeRegion,
-        SpatialRange,
-        TimeRange,
-    )
-    from virtualship.expedition.waypoint import Location, Waypoint
+        if ext == ".csv":
+            return pd.read_csv(file_path)
+
+        raise ValueError(f"Unsupported file extension {ext}.")
 
+    except Exception as e:
+        raise RuntimeError(
+            "Could not read coordinates data from the provided file. "
+            "Ensure it is either a csv or excel valid file."
```

Comment:
Not sure what "valid" means in this context...
```suggestion
            "Ensure it is either a csv or excel file."
```